### PR TITLE
chore: bump driver version to v0.0.18

### LINKED ISSUE #398: Update ldflags to use new build variables
https://github.com/kubernetes-sigs/secrets-store-csi-driver/pull/396 adds `BuildVersion`, `BuildTime` and `Vcs` env vars to inject build time variables. With this we can remove the hard coding of `vendorVersion` in the `secrets-store` package.

Update the build scripts in `docker/*` dir to use the new ldflags as part of the next release. This change needs to be done during the release because any change to `docker/*` dir will trigger the post submit release job to build a new image.

/kind cleanup
/assign

### PR DESCRIPTION
**Pull request by aramase**
**What this PR does / why we need it**:
- Bumps driver version to `v0.0.18` for release
- Adds ldflags changes

**Which issue(s) this PR fixes** *(optional, using `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when the PR gets merged)*:
Fixes #398

**Special notes for your reviewer**:

### CODE REVIEW

**Comment by aramase** [comment]
/test pull-secrets-store-csi-driver-e2e-windows
/hold

(wait for windows tests to complete)

File: docker/build.sh [hunk 1]
```diff
@@ -71,7 +73,10 @@ docker_version_check() {
 build_and_push() {
   docker_version_check
 
-  docker buildx create --name img-builder --use
+	# only moby/buildkit:foreign-mediatype works on building Windows image now
+	# https://github.com/moby/buildkit/pull/1879
+	# Github issue: https://github.com/moby/buildkit/issues/1877
```
Comment (tam7t) [inline docker/build.sh:78]: nit: i think this is using tabs when rest of file is spaces

**Review by tam7t — CHANGES_REQUESTED** [review]

File: docker/build.sh (hunk 1 above)
Comment (aramase) [reply docker/build.sh]: thank you. It should be updated now.

**Pull request merged**